clcache.py

```diff
@@ -126,6 +128,80 @@ def release(self):
         self._acquired = False
 
 
+class ObjectCacheLockFile:
+    """ Implements a lock file lock for the object cache which
+    can be used in 'with' statements.
+    Inspired by
+    https://github.com/harlowja/fasteners
+    https://github.com/benediktschmitt/py-filelock"""
+
+    def __init__(self, lockfileName, timeoutMs):
+        self._lockfileName = lockfileName
+        self._lockfile = None
+        self._timeoutMs = timeoutMs
+
+    def __enter__(self):
+        self.acquire()
+
+    def __exit__(self, typ, value, traceback):
+        self.release()
+
+    def __del__(self):
+        self.release()
+
+    def _acquire(self):
+        try:
+            lockfile = open(self._lockfileName, 'a')
+        except OSError:
+            lockfile = None
+        else:
+            try:
+                msvcrt.locking(lockfile.fileno(), msvcrt.LK_NBLCK, 1)
+            except (IOError, OSError):
+                lockfile.close()
+            else:
+                self._lockfile = lockfile
+        return None
```

Comment:
For [stylistic reasons|http://stackoverflow.com/questions/15300550/python-return-return-none-and-no-return-at-all], I think it would be fine to just emit the `return` statement completely here: the method never returns any value in any of the code paths.